### ISSUE TITLE
Convert examples to `fn main() -> GameResult<()>`

### ISSUE THREAD

**Issue by ozkriff** [ggez/ggez#382]
Maybe it's time to upgrade the examples and remove all the expect/unwrap stuff from `main`? I think it'll result in a shorter and cleaner code:

```rust
pub fn main() {
    let c = conf::Conf::new();
    let ctx = &mut Context::load_from_conf("super_simple", "ggez", c).unwrap();
    let state = &mut MainState::new(ctx).unwrap();
    event::run(ctx, state).unwrap();
}
```

vs

```rust
fn main() -> GameResult<()> {
    let c = conf::Conf::new();
    let ctx = &mut Context::load_from_conf("super_simple", "ggez", c)?;
    let state = &mut MainState::new(ctx)?;
    event::run(ctx, state)
}
```

------

Btw (1), why `pub fn main` and not just `fn main`?

Btw (2), what is GGEZ's minimal required Rust version? It may be a good idea to explicitly say that in readme ([using badge?](https://img.shields.io/badge/rust-1.25+-blue.svg?label=Required%20Rust) :)) and use that version in `.travis.yml`.

**Comment by icefoxen** [comment]
It does look nicer, I have to agree. Not sure why the `pub` is there, to be honest.

Minimum rustc version was actually just recently figured out, see #376

**Comment by ozkriff** [comment]
> Minimum rustc version was actually just recently figured out, see #376. What version was Result in main stabilized?

Sorry, I was looking at master branch.

> It does look nicer, I have to agree. Not sure why the pub is there, to be honest.

Then I'll make a PR, if you don't mind?

**Comment by Ratysz** [comment]
`Result` in `main()` is stable since Rust 1.26.

I like this too, it's perfect for brief and clean examples (although for bigger executable projects I'd keep using the "`main()` as thin error handler" pattern).

**Comment by icefoxen** [comment]
Because my reflex is to be conservative with Rust versions, would anyone mind waiting a bit before making this change? Just so 1.26 becomes a bit more widespread.

Perhaps we can make this part of the ggez 0.5 release, since that will be a large change anyway.

**Comment by icefoxen** [comment]
Resolved in f992cd7
